src/lib/utils/router.js

```diff
@@ -99,7 +96,7 @@ export function listen(handler) {
   if (!handler) {
     throw new Error('need handler');
   }
-  listen = () => {
+  listen = () => { // eslint-disable-line
```

Comment:
@robdodson @samthor How is listen being reassigned as a function?

Reply:
Sam can correct me. I think this might be some hecka clever JS. Basically it's reassigning the parent function to itself, then running to completion so its side effects still happen. After that if you call listen() ever again it will throw.

Reply:
@MichaelSolati I think we may want to remove the TODO on line 90 and replace it with an explanation of how this thing works.

Reply:
Also Sam should double check that my assessment is correct :)